subsys/caf/modules/ble_adv.c

```diff
@@ -131,27 +124,61 @@ static void conn_find_cb(struct bt_conn *conn, void *data)
 	(*temp_conn) = conn;
 }
 
-static void bond_find(const struct bt_bond_info *info, void *user_data)
+static inline struct bt_conn *conn_get(void)
+{
+	struct bt_conn *conn = NULL;
+
+	bt_conn_foreach(BT_CONN_TYPE_LE, conn_find_cb, &conn);
```

Comment:
Are you aware that this iterator can potentially return the disconnected or connecting (advertising) object?

Reply:
Yes and in that case we still cannot start the advertising (that's what for the conn check is used)